clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/view/model/KubernetesV2Cluster.java

```diff
@@ -38,12 +38,14 @@
   String accountName;
   Set<ServerGroup> serverGroups = new HashSet<>();
   Set<LoadBalancer> loadBalancers = new HashSet<>();
+  String application;
 
   public KubernetesV2Cluster(String rawKey) {
     Keys.ClusterCacheKey key = (Keys.ClusterCacheKey) Keys.parseKey(rawKey).get();
     this.name = key.getName();
     this.accountName = key.getAccount();
-    this.moniker = Moniker.builder().cluster(name).build(); // TODO(lwander) if it turns out that cluster -> app is important, enforce constraints here.
+    this.application = key.getApplication();
+    this.moniker = Moniker.builder().cluster(name).app(application).build(); // TODO(lwander) if it turns out that cluster -> app is important, enforce constraints here.
```

Comment:
Looks like` cluster -> app` might become important :)

Reply:
Haha so prescient of you! Lmk if there are any constraints in particular you'd like me to add with this change.

Reply:
No I think you solved them by allowing the clusters to belong to multiple projects. You can delete the TODO